server/routes/creatingReports/whyWasUoFApplied.ts

```diff
@@ -5,52 +5,79 @@ import { paths } from '../../config/incident'
 
 const FORM = 'reasonsForUseOfForce'
 
+enum SubmitType {
+  SAVE_AND_CONTINUE = 'save-and-continue',
+  SAVE_AND_RETURN = 'save-and-return',
+}
+
 export default class WhyWasUoFAppliedRoutes {
   constructor(private readonly draftReportService: DraftReportService) {}
 
   public view() {
     return async (req: Request, res: Response): Promise<void> => {
       const { bookingId } = req.params
-      const { reasons } = await this.draftReportService.getSelectedReasonsForUoF(req.user.username, Number(bookingId))
+      const { isComplete, reasons } = await this.draftReportService.getUoFReasonState(
+        req.user.username,
+        Number(bookingId)
+      )
+
+      const selectedReasons = req.flash('clearingOutReasons')?.length ? [] : reasons
 
       return res.render('formPages/incident/select-uof-reasons', {
         errors: req.flash('errors'),
-        data: { UofReasons, reasons },
+        data: { bookingId, UofReasons, reasons: selectedReasons },
+        editMode: isComplete,
       })
     }
   }
 
   public submit(): RequestHandler {
     return async (req, res) => {
       const { bookingId } = req.params
-      const { reasons } = req.body
+      const { reasons, submitType } = req.body
 
       if (!reasons) {
+        req.flash('clearingOutReasons', true)
         req.flash('errors', [{ href: '#reasons', text: 'Select the reasons why use of force was applied' }])
         return res.redirect(req.originalUrl)
       }
 
-      await this.draftReportService.process(res.locals.user, Number(bookingId), FORM, { reasons })
+      if (submitType === SubmitType.SAVE_AND_RETURN) {
+        await this.draftReportService.process(res.locals.user, Number(bookingId), FORM, { reasons })
+        return res.redirect(paths.reportUseOfForce(bookingId))
+      }
 
-      return reasons.length > 1
-        ? res.redirect(paths.whatWasPrimaryReasonForUoF(bookingId))
-        : res.redirect(paths.useOfForceDetails(bookingId))
+      if (reasons.length === 1) {
+        await this.draftReportService.process(res.locals.user, Number(bookingId), FORM, { reasons })
+        return res.redirect(paths.useOfForceDetails(bookingId))
+      }
+
+      req.flash('reasons', reasons)
+      return res.redirect(paths.whatWasPrimaryReasonForUoF(bookingId))
     }
   }
 
   public viewPrimarySelection() {
     return async (req: Request, res: Response): Promise<void> => {
       const { bookingId } = req.params
 
-      const { reasons, primaryReason } = await this.draftReportService.getSelectedReasonsForUoF(
+      const { primaryReason, reasons } = await this.draftReportService.getUoFReasonState(
         req.user.username,
         Number(bookingId)
       )
-      const selectedReasons = Object.values(UofReasons).filter(({ value }) => reasons.includes(value))
+
+      const updatedReasons = req.flash('reasons')
+      const reasonsToUse = updatedReasons.length ? updatedReasons : reasons
```

Comment:
To cope with backwards navigation from the use of force details, we present the user the persisted values (as there will be no flash content).

This has the side effect of meaning that If a user refreshes the screen (clears out flash) then they will be presented the old persisted values again.

Seems like an ok trade off as users are probably more likely to use backwards navigation compared to refreshing